realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -65,39 +99,7 @@ public void generate() throws IOException, UnsupportedOperationException {
         writer.emitPackage(Constants.REALM_PACKAGE_NAME)
                 .emitEmptyLine();
 
-        ArrayList<String> imports = new ArrayList<String>();
-        imports.add("android.annotation.TargetApi");
-        imports.add("android.os.Build");
-        imports.add("android.util.JsonReader");
-        imports.add("android.util.JsonToken");
-        imports.add("io.realm.RealmObjectSchema");
-        imports.add("io.realm.RealmSchema");
-        imports.add("io.realm.exceptions.RealmMigrationNeededException");
-        imports.add("io.realm.internal.ColumnInfo");
-        imports.add("io.realm.internal.RealmObjectProxy");
-        imports.add("io.realm.internal.Row");
-        imports.add("io.realm.internal.Table");
-        imports.add("io.realm.internal.OsObject");
-        imports.add("io.realm.internal.SharedRealm");
-        if (!metadata.getBacklinkFields().isEmpty()) {
```

Comment:
I'm fine with adding @SuppressWarnings("all") at the top of the file if it works, but e.g ErrorProne will catch warnings in the generated files, so it is requirement that we don't have any. There was talk about adding @Generated annotation as well at some point, but afaik that isn't available on Android.